GoogleForCreators/web-stories-wp
# Custom fonts: do not re-fetch fonts after deletion

## Context



## Summary

When adding and removing custom fonts, another REST API request is made after the add / remove request is made. This is to get all fonts. This API request is not needed, as the font data is stored in state. After the request is complete, simple remove / add the font to the array. 

## Relevant Technical Choices

Refactor some code to be a better async, while we are changing this code. 

## To-do



## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

1. Go to dashboard > Settings.
2. Add a custom font. Eg [overpass-extrabold-italic.ttf](https://overpass-30e2.kxcdn.com/overpass-extrabold-italic.ttf)
3. See new font to added straight away. 
4. Try to add some font again. 
5. See error message. 
6. Delete font.
7. See font removed straight away. 


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #10940
